### PR TITLE
android support

### PR DESCRIPTION
**Pull request by pazos**
1. restore custom light at launch
2. store last custom level used to repurpose it on next launch
3. return back to system settings each time we check the button.

### CODE REVIEW

**Comment by pazos** [comment]
BTW,  it requires https://github.com/koreader/android-luajit-launcher/pull/171

File: frontend/device/android/device.lua [hunk 1]
```diff
@@ -195,6 +195,12 @@ function Device:init()
         android.setVolumeKeysIgnored(true);
     end
 
+    -- check if we enable a custom light level for this activity
+    local last_value = G_reader_settings:readSetting("last_level")
```
Comment (robert00s) [inline frontend/device/android/device.lua]:
```suggestion
    local last_value = G_reader_settings:readSetting("fl_last_level")
```

File: frontend/ui/widget/frontlightwidget.lua [hunk 2]
```diff
@@ -139,6 +139,14 @@ function FrontLightWidget:setProgress(num, step, num_warmth)
             else
                 self.powerd:setIntensity(set_fl)
             end
+
+            if not self.light_fallback and self.fl_cur >= 0 then
+                G_reader_settings:saveSetting("last_level", self.fl_cur * 10)
```
Comment (robert00s) [inline frontend/ui/widget/frontlightwidget.lua]:
```suggestion
                G_reader_settings:saveSetting("fl_last_level", self.fl_cur * 10)
```

File: frontend/ui/widget/frontlightwidget.lua [hunk 3]
```diff
@@ -139,6 +139,14 @@ function FrontLightWidget:setProgress(num, step, num_warmth)
...
+            elseif self.light_fallback then
+                G_reader_settings:saveSetting("last_level", nil)
```
Comment (robert00s) [inline frontend/ui/widget/frontlightwidget.lua]:
```suggestion
                G_reader_settings:saveSetting("fl_last_level", nil)
```

**Review by robert00s — COMMENTED** [review]
`fl_last_level` looks better than `last_level`

**Comment by robert00s** [comment]
@pazos Could you commit my small suggestions? 
Other things looks good to me :)

**Comment by pazos** [comment]
@robert00s: done. Bumped luajit-launcher too!

**Comment by robert00s** [comment]
Thanks :)

**Pull request merged**